timdex_dataset_api/dataset.py

```diff
@@ -371,14 +376,18 @@ def read_batches_iter(
         Args:
             - table: an available DuckDB view or table
             - columns: list of columns to return
+            - limit: limit number of records yielded
             - where: raw SQL WHERE clause that can be used alone, or in combination with
             key/value DatasetFilters
             - filters: simple filtering based on key/value pairs from DatasetFilters
         """
+        start_time = time.perf_counter()
+
         # build and execute metadata query
         metadata_time = time.perf_counter()
-        meta_query = self.metadata.build_meta_query(table, where, **filters)
+        meta_query = self.metadata.build_meta_query(table, limit, where, **filters)
         meta_df = self.metadata.conn.query(meta_query).to_df()
+        meta_df = meta_df.sort_values(by=["filename", "run_record_offset"])
```

Comment:
This was another fairly big improvement, and an interesting one.

Instead of sorting the metadata results in DuckDB via `ORDER BY`, we complete the query unsorted but then do that here in python + pandas.  

I anticipate a time in the not-too-distant future where we take another pass at this code to perform our metadata queries in batches.  If we do that, maybe using `LIMIT` + `OFFSET`, then we might need to move the sort back into DuckDB.  

But for now, this is a clear performance gain with minimal change, just changing where we do the sort.